harness/determined/common/experimental/checkpoint/_checkpoint.py

```diff
@@ -449,6 +449,23 @@ def get_metrics(self, group: Optional[str] = None) -> Iterable["metrics.TrialMet
         for d in resp.metrics:
             yield metrics.TrialMetrics._from_bindings(d, group)
 
+    def get_pachyderm_commit(self) -> str:
```

Comment:
non-blocking question: Is there a reason why we're exposing the commit hash specifically? Is there a use case for retrieving this field that's different than the other fields in the pachyderm configuration?

Reply:
yeah, it's mainly a product framing consideration. we want to make the integration seem smooth. 

conceptually we're putting pachyderm integration on a determined checkpoint assuming the user wants to get the data associated with this checkpoint. my understanding is that the commit hash is the only thing you need to be able to get the data in pachyderm, so the idea here is to make the user journey between determined checkpoints and its pachyderm data obvious. it'd be weird if a determined checkpoint object returned the whole pachyderm config.